xbmc/interfaces/legacy/InfoTagVideo.h

```diff
@@ -9,606 +9,2882 @@
+#endif
+
+    private:
+      int m_channels;
+      String m_codec;
+      String m_language;
+    };
+
+    ///
+    /// \defgroup python_xbmc_subtitlestreamdetail SubtitleStreamDetail
+    /// \ingroup python_xbmc
+    /// @{
+    /// @brief **Subtitle stream details class used in combination with InfoTagVideo.**
+    ///
+    /// \python_class{ xbmc.SubtitleStreamDetail([language]) }
+    ///
+    /// Represents a single selectable subtitle stream for a video item wrapped by InfoTagVideo.
+    ///
+    ///
+    ///-------------------------------------------------------------------------
+    /// @python_v20 New class added.
+    ///
+    ///
+    /// **Example:**
+    /// ~~~~~~~~~~~~~{.py}
+    /// ...
+    /// subtitlestream = xbmc.SubtitleStreamDetail('English')
+    /// ...
+    /// ~~~~~~~~~~~~~
+    ///
+    class SubtitleStreamDetail : public AddonClass
+    {
+    public:
+#ifdef DOXYGEN_SHOULD_USE_THIS
+      ///
+      /// \ingroup python_xbmc_subtitlestreamdetail SubtitleStreamDetail
+      /// @brief \python_func{ xbmc.SubtitleStreamDetail([language]) }
+      /// Creates a single subtitle stream details class for a video item wrapped by InfoTagVideo.
+      ///
+      /// @param language           [opt] string - Language of the subtitle.
+      ///
+      ///
+      ///-----------------------------------------------------------------------
+      ///
+      /// **Example:**
+      /// ~~~~~~~~~~~~~{.py}
+      /// ...
+      /// subtitlestream = xbmc.SubtitleStreamDetail('English')
+      /// ...
+      /// ~~~~~~~~~~~~~
+      ///
+      SubtitleStreamDetail(...);
+#else
+      explicit SubtitleStreamDetail(const String& language = emptyString);
+#endif
+
+#ifdef DOXYGEN_SHOULD_USE_THIS
+      ///
+      /// \ingroup python_xbmc_subtitlestreamdetail
+      /// @brief \python_func{ getLanguage() }
+      /// Get the language of the stream.
+      ///
+      /// @return [string] Language of the stream
+      ///
+      ///
+      ///-----------------------------------------------------------------------
+      ///
+      getLanguage();
+#else
+      String getLanguage() const { return m_language; }
+#endif
+
+#ifdef DOXYGEN_SHOULD_USE_THIS
+      ///
+      /// \ingroup python_xbmc_subtitlestreamdetail
+      /// @brief \python_func{ setLanguage(language) }
+      /// Set the language of the stream.
+      ///
+      /// @param language           string - Language of the stream.
+      ///
+      ///
+      ///-----------------------------------------------------------------------
+      ///
+      setLanguage(...);
+#else
+      void setLanguage(const String& language) { m_language = language; }
+#endif
+
+#ifndef SWIG
+      CStreamDetailSubtitle* ToStreamDetailSubtitle() const;
+#endif
+
+    private:
+      String m_language;
+    };
+
+    ///
+    /// \defgroup python_xbmc_fanart Fanart
+    /// \ingroup python_xbmc
+    /// @{
+    /// @brief **Fanart class used in combination with InfoTagVideo.**
```

Comment:
Well fanart seems to be handled differently in core than any other media type. I don't really know why because at least thumbs / posters should work the same way.

Reply:
See my previous comment.

Reply:
Removed new fanart class and interface.

Reply:
The "specialness" of fanart is historic (once there was only thumbs and fanart). @rmrector and I have been slowly working on making it like everything else but some unique handling still remains. However that is internally, there is no reason for special treatment when offering Python access.